File: Y2S2/SDI/lab9-11/packageManagerLab10/core/src/main/java/ro/arc/packageManager/core/service/PackageVersionService.java
package ro.arc.packageManager.core.service;

import ro.arc.packageManager.core.domain.PackageVersion;

import java.util.List;

public interface PackageVersionService {
    PackageVersion addPackageVersion(PackageVersion packageVersion);

    PackageVersion updatePackageVersion(PackageVersion packageVersion);

    void deletePackageVersion(Long ID);

    List<PackageVersion> getFilteredPackageVersions(String type, String input);

    List<PackageVersion> getAllPackageVersions();
}

File: Y2S2/SDI/lab9-11/packageManagerLab11/web/src/main/webapp/src/app/package-versions/shared/packageVersion.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { PackageVersionModel } from './packageVersion.model';

@Injectable()
export class PackageVersionService {
  private packageMaintainersUrl = 'http://localhost:8080/web-1.0-SNAPSHOT/api/packageVersions';

  constructor(private httpClient: HttpClient) { }

  getPackageVersions(packageID: number): Observable<PackageVersionModel[]> {
    return this.httpClient.get<Array<PackageVersionModel>>(`${this.packageMaintainersUrl}/filter/packageID=${packageID}`);
  }

  addPackageVersion(entity: PackageVersionModel): Observable<PackageVersionModel> {
    return this.httpClient.post<PackageVersionModel>(this.packageMaintainersUrl, entity);
  }

  updatePackageVersion(entity: PackageVersionModel): Observable<PackageVersionModel> {
    return this.httpClient.post<PackageVersionModel>(`${this.packageMaintainersUrl}/${entity.packageID}`, entity);
  }

  deletePackageVersion(entity: PackageVersionModel): Observable<any> {
    return this.httpClient.delete(`${this.packageMaintainersUrl}/${entity.packageID}`);
  }
}
